packages/app/src/components/common/CodeClimbersLoadingButton.tsx

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable codeclimbers/use-code-climbers-button */
+// eslint-disable-next-line import/no-named-as-default
+import posthog from 'posthog-js'
+import { LoadingButton, LoadingButtonProps } from '@mui/lab'
+
+type Props = LoadingButtonProps & {
+  eventName: string
+  target?: string
+}
+
+const CodeClimbersLoadingButton = ({
```

Comment:
Wouldn't it just be better to have a generic button component that takes in the eventName? LoadingButton to me mean that you'd be displaying a spinner and expect a `isLoading | isPending` prop